### PR TITLE
feat(install): provide redirect URL for fallback installation, add installation summary

### PR DESCRIPTION
**Pull request by sanderblue**
Provides a redirect URL after an incomplete installation process that allows the user to return to the UI and attempt to finish incomplete installations.

This also provides an installation summary similar to what's displayed in the UI at the end of the installation process.

### CODE REVIEW

File: internal/install/execution/platform_link_generator.go
Comment (Julien4218) [inline internal/install/execution/platform_link_generator.go:49]: @sanderblue do we need to send any filter, hostname in the URL? if not  the URL should be simpler without the `platform[filters]=[hostname]` parts.
Maybe we should have a different method specific to this use case.

Also, wouldn't we want to have a same URL for success? We essentially want the user to go back to the NR1 Installation Plan page regardless of success/failure/cancel?

Comment (sanderblue) [reply internal/install/execution/platform_link_generator.go:49]: We need the `platform[filters]` because of the `successLinkConfig` parameter in recipes.

Regarding when we add the referrer param: I was asking myself the same question. I think we can append this param in all situation and let the UI decide what it does with it.

Comment (sanderblue) [reply internal/install/execution/platform_link_generator.go:49]: Making some updates. Will re-request review tomorrow.

File: internal/install/command.go
Comment (Julien4218) [inline internal/install/command.go:67]: @sanderblue I think that whole phrasing should be reviewed (changed), I dont think we should say "error".
Check with Sam/Aaron, perhaps more something like "We couldn't complete this installation, details :%s" and send the user back to the UI.

File: internal/install/execution/terminal_reporter.go
Comment (Julien4218) [inline internal/install/execution/terminal_reporter.go:137]: @sanderblue we should check the color coding, Red might be viewed as too harsh (making users think of an error), maybe yellow, worth checking with Sam/Aaron and/or Daniel

File: internal/install/command.go
Comment (sanderblue) [reply internal/install/command.go:67]: Agreed. I mentioned it to Sam. In general the output needs additional organization to make it friendlier to the users' eye.

File: internal/install/recipe_installer.go
Comment (Julien4218) [inline internal/install/recipe_installer.go:352]: @sanderblue another "Failed" message here needing to be updated.

File: internal/install/command.go
Comment (Julien4218) [reply internal/install/command.go:67]: I assume you'll reach out to him to finalize those details? Let me know when done and I'll start testing.

Comment (sanderblue) [reply internal/install/command.go]: I'll confirm tomorrow.

File: internal/install/recipe_installer.go
Comment (sanderblue) [reply internal/install/recipe_installer.go:352]: Do we care about that for debug messages vs default user-facing messages?

Comment (Julien4218) [reply internal/install/recipe_installer.go:352]: Ho sorry I didnt see this was debug and not the default output for users, that's fine then.

File: internal/install/command.go
Comment (sanderblue) [reply internal/install/command.go]: Closing the loop, we're changing this to a debug log since the general output points the customer back to finishing any incomplete installations. As far as I know, we only have one scenario where we run into this error and that's when the recipes are not compiled, so likely will only occur during local dev. I've accounted for both of these scenario 😎

File: internal/install/execution/terminal_reporter.go
Comment (sanderblue) [reply internal/install/execution/terminal_reporter.go]: Updated to yellow and a less harsh icon :)

**Comment by Julien4218** [comment]
@sanderblue I've reached out in the team channel regarding one of the user message than I found misleading.
Also when I run the test local, I see this test failing, any idea why?
=== FAIL: internal/install/execution TestGenerateRedirectURL_InstallSuccess (0.00s)
    platform_link_generator_test.go:41:
                Error Trace:    platform_link_generator_test.go:41
                Error:          Not equal:
                                expected: "https://staging-one.newrelic.com/redirect/entity/ABC123"
                                actual  : "https://one.newrelic.com/redirect/entity/ABC123"

**Comment by Julien4218** [comment]
Test is now fixed, thanks @sanderblue

**Review by Julien4218 — CHANGES_REQUESTED** [review]
@sanderblue, I'm holding this work until we have the green light to ship, since the UI will be behind FF but this work isn't (we discussed in planning not needing to this at the CLI level).

File: internal/install/execution/platform_link_generator_integration_test.go
Comment (ctrombley) [inline internal/install/execution/platform_link_generator_integration_test.go:53]: nit: remove `&ab_channel=RickAstley` from the URL so as not to give away the punchline

**Review by Julien4218 — APPROVED** [review]

**Pull request merged**
